ENH: Support for LSF clusters (#268)

juanprietob opened the pull request.

juanprietob committed:
ENH: Support for LSF clusters

glatard commented on BrainPortal/lib/scir_lsf.rb:
```diff
@@ -28,73 +28,62 @@ class ScirLsf < Scir
-      out, err = bash_this_and_capture_out_err("bstat -f -u #{CBRAIN::Rails_UserName.to_s.bash_escape}")
-      raise "Cannot get output of 'bstat -f' ?!?" if out.blank? && ! err.blank?
+      out, err = bash_this_and_capture_out_err("bjobs -a -noheader")
+      raise "Cannot get output of 'bjobs -a' ?!?" if out.blank? && ! err.blank?
```
Can't you filter the jobs that belong to a particular user, as it is done with option -u in qsub? Otherwise it will put a lot of unnecessary stuff in memory when clusters are shared with other users.

juanprietob replied:
Yes, I did not see the option before. 
`-J <job_name_spec>`

glatard commented:
Super! Did you have a chance to test it, or does it still need to be tested?

Commit:
BUG: The jobs must be listed for the user

The jobs are listed using Display the jobs specified by job_name_spec. job_name_spec is

juanprietob commented:
I need to test it with multiple users.

glatard commented on BrainPortal/lib/scir_lsf.rb:
```diff
@@ -28,73 +28,62 @@ class ScirLsf < Scir
       # the @job_info_cache array.
-      out, err = bash_this_and_capture_out_err("bstat -f -u #{CBRAIN::Rails_UserName.to_s.bash_escape}")
-      raise "Cannot get output of 'bstat -f' ?!?" if out.blank? && ! err.blank?
+      out, err = bash_this_and_capture_out_err("bjobs -a -noheader -J #{shell_escape(self.name)}")
```
What about adding option -u to bjobs: `bjobs -a -noheader -J #{shell_escape(self.name)} -u #{CBRAIN::Rails_UserName.to_s.bash_escape}`?

glatard commented:
Hi @juanprietob, any news about that?

Commit:
BUG: Remove ending ':'

Remove ending ':' and use different flag to set the stdout

juanprietob commented:
I have one more issue. 
The script is not found when launching the job. 

This is the command executed to lanuch the job.

`bsub -J 'MyCksum-36' -u 'MyCksum-36' -cwd '/lustre/scr/j/p/jprieto/shared/00/00/36/admin-MyCksum-T36' -o '/lustre/scr/j/p/jprieto/shared/00/00/36/admin-MyCksum-T36/.qsub.out.MyCksum.36-1' -e '/lustre/scr/j/p/jprieto/shared/00/00/36/admin-MyCksum-T36/.qsub.err.MyCksum.36-1' -q 'week' '.qsub.MyCksum.36-1.sh' 2>&1`

However, the script `.qsub.MyCksum.36-1.sh` is not found. 
Should I export the `CWD` into the env `PATH`?

glatard commented:
No, you shouldn't touch the PATH. Is the script created in the right directory? Maybe it also need to have execution permissions. 

You could also try to 'cd' in the task execution directory and launch the bsub command from there, to see how the issue should be solved in the bsub command (before fixing this in CBRAIN).

juanprietob commented:
Yes, the script is created in the right directory. 

Do you mean add `cd CWD; bsub ...`?
